app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,81 @@
+class TweetsController < ApplicationController
+
+  get '/tweets' do
+    if !logged_in?
+      redirect '/login'
+    else
+      @user = User.find_by_id(session[:user_id])
+      @tweets = Tweet.all
+      erb :'tweets/tweets'
+    end
+  end
+
+  get '/test' do
+    @user = User.find_by_id(session[:user_id])
+    @tweets = Tweet.all
+    erb :'users/show'
+  end
+
+  get '/edit' do
+    if !logged_in?
+      redirect '/login'
+    else
+      erb :'tweets/edit_tweet'
+    end
+  end
+
+  get '/tweets/new' do
+    if !logged_in?
+      redirect '/login'
+    else
+      erb :'tweets/create_tweet'
+    end
+  end
+
+  post '/new' do
+    if !logged_in?
+      redirect '/login'
+    elsif
+      params[:content].empty?
+      redirect '/tweets/new'
+    else
+      @user = current_user
+      @user.tweets.build(content: params[:content])
+      @user.save
+      redirect '/tweets'
+    end
+  end
+
+  get '/tweets/:id' do
+    if !logged_in?
+      redirect "/login"
+    else
+      @tweet = Tweet.find(params[:id])
+      erb :"tweets/show_tweet"
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if !logged_in?
+      redirect to "/login"
+    else
+      @tweet = Tweet.find(params[:id])
+      erb :"tweets/edit_tweet"
+    end
+  end
+
+  patch '/tweets/:id/edit' do
+    tweet = Tweet.find(params[:id])
+    tweet.content = params[:content]
+    tweet.save
+  end
+
+  delete '/tweets/:id/delete' do
+    tweet = Tweet.find(params[:id])
+    if logged_in? && tweet.user_id == current_user.id
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.